r/tests/testthat/test-dplyr-lubridate.R

```diff
@@ -0,0 +1,160 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+library(lubridate)
+library(dplyr)
+
+test_date <- as.POSIXct("2017-01-01 00:00:12.3456789", tz = "")
+test_df <- tibble::tibble(date = test_date)
+
+tz_aware_date <- ymd_hms("2017-01-01 00:00:12.3456789")
+tz_aware_df <- tibble::tibble(date = tz_aware_date)
+
+test_that("timezone aware timestamps are not supported",{
+  x <- Expression$field_ref("x")
+  expect_error(
+    Table$create(tz_aware_df) %>%
+      mutate(x = wday(date)) %>%
+      collect(),
+    "Timezone aware timestamps not supported"
+  )
+})
```

Comment:
I added in the above test to demonstrate the current/expected behaviour, but I'm not totally happy with this.  I'm not sure what makes sense to do here.

I need to properly catch up on the mailing list discussion on timezone aware timestamps to work out the likely possibilities and solutions, so will come back to this, but not happy with it as-is at the moment.

Reply:
Not entirely sure, but I think when this is merged, it will allow the above issue to be resolved: https://issues.apache.org/jira/browse/ARROW-12980

Reply:
I think this is fine for now, since we this error on purpose in the initial PR to limit scope. And this functionality will indeed be added in a next PR.